indico/modules/rb_new/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -0,0 +1,419 @@
+            case 'checkbox':
+                return (
+                    <Field key={content.name}
+                           name={content.name}
+                           component={ReduxCheckboxField}
+                           componentLabel={content.label}
+                           as={Checkbox} />);
+            case 'image':
+                return this.renderImage(room.spritePosition);
+            case 'attributes':
+                return this.renderAttributes();
+            case 'addButton':
+                return this.renderAddButton(content.editType);
+            case 'dropdown':
+                if (!attributes) {
+                    return;
+                }
+                options = this.generateOptions(attributes);
+                return (
+                    <Dropdown placeholder="Add new attributes"
+                              fluid
+                              multiple
+                              selection
+                              options={options}
+                              onChange={(event, values) => this.setState({newAttributes: values.value})} />
+                );
+        }
+    };
+
+
+    renderModalContent = (fprops) => {
+        const formProps = {onSubmit: fprops.handleSubmit};
+        return (
+            <>
+                <Modal.Header>
+                    <Translate>Edit Room Details</Translate>
+                </Modal.Header>
+                <Modal.Content>
+                    <Form id="room-form" {...formProps}>
+                        <Grid columns={3}>
+                            {columns.map(this.renderColumn)}
+                        </Grid>
+                    </Form>
+                </Modal.Content>
+                <Modal.Actions>
+                    <Button type="submit"
+                            form="room-form"
+                            primary>
+                        <Translate>Save</Translate>
+                    </Button>
+                </Modal.Actions>
+            </>
+        );
+    };
+
+    addAttributes = () => {
+        const {roomAttributes, attributes, newAttributes} = this.state;
+
+        const attributesObj = _.keyBy(attributes, 'name');
+        let newRoomAttributes = {};
+        newAttributes.map((attr) => {
+            newRoomAttributes = {...{[attributesObj[attr].name]: null}, ...newRoomAttributes};
+            return newRoomAttributes;
+        });
+        this.setState({roomAttributes: {...roomAttributes, ...newRoomAttributes}});
+    };
+
+    generateOptions = (attributes) => {
+        const {roomAttributes} = this.state;
+        const options = [];
+        attributes.map((key) => (!roomAttributes.hasOwnProperty(key.name)
+            ? options.push({key: key.name, text: key.title, value: key.name}) : null));
+        return options;
+    };
+
+
+    render() {
+        const {room, roomAttributes, attributes} = this.state;
+        const props = {validate, onSubmit: this.handleSubmit};
```

Comment:
you always have the same props, so just pass them directly in the `<FinalForm>` tag